ConfCore/Bookmark.swift

```diff
@@ -57,4 +57,33 @@ public final class Bookmark: Object, HasCloudKitFields, SoftDeletable {
         return "identifier"
     }
 
+    // MARK: - Codable
+
+    private enum CodingKeys: String, CodingKey {
+        case body
+        case attributedBody
+        case timecode
+        case duration
+    }
+
+    public required convenience init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        let body = try container.decode(String.self, forKey: .body)
+        let timecode = try container.decode(Double.self, forKey: .timecode)
+        let duration = try container.decode(Double.self, forKey: .duration)
+
+        self.init()
```

Comment:
@insidegui I'm curious what your thoughts are on setting the decoding up like this. I see that the purpose is to avoid object initialization until it has been determined that an object is needed, i.e. decoding succeeded, but in my opinion it seems to be optimizing for what * should * be the less likely scenario (most of the objects will decode successfully) and makes the code a bit longer and unwieldy.

Reply:
I'm sorry, I don't know what exactly it is you're referring to, to me this just looks like a normal custom `init` for a `Decodable` type 😅

Reply:
Compared to this https://github.com/insidegui/WWDC/pull/530/files#diff-480a5a40eae0b47ad936d50c14cd5346R67

I rewrote it so `self.init` is called first, which means if the decoding fails, memory will be allocated and then deallocated but there is no need for the local variables

Reply:
Oh, I see. Yeah, I don't see any problem with that.